### PR TITLE
docs: Add demo to homepage

### PR DESCRIPTION
**Pull request by samestep**
# Description

This PR adds a new `Demo` component which cycles through a fixed collection of trios/variations (changing every 5 seconds), showing the Substance program and animating the diagram's convergence at 60 fps.

# Implementation strategy and design decisions

We expected that it would be straightforward to embed a React component in the Docusaurus site, but it was not. @wodeni and @maxkrieger put in a lot of effort to get this working, including replacing the build systems for a few of our packages. This also involved an upgrade to (among other dependencies) Docusaurus itself, which required us to update the filenames of some of our images from the tutorials.

Also, as part of this PR, I essentially completely rewrote the `Simple` Penrose diagram component. Now it has more features and also implements its React state management in a somewhat more idiomatic way.

# Checklist

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new ESLint warnings
- [x] I have reviewed any generated changes to the `diagrams/` folder

# Open questions

The `Demo` component UI is still pretty barebones:

- its size on the Docusaurus site is defined in terms of pixels
- it doesn't have any buttons to navigate among the different examples
- it doesn't give any way to look at the underlying Domain or Style programs
- it doesn't give any visual indication of the 5-second countdown

@joshsunshine suggested that perhaps @hsharriman might be able to help improve this UI in the future.

Also, we need some spicier diagrams to actually show in the `Demo` component on the homepage. Perhaps @keenancrane has some up his sleeve?

### CODE REVIEW

**Comment by maxkrieger** [comment]
btw I disabled netlify because they greedily tried to bill us just for using the free version - we'll want to switch to cloudflare pages (and nestle panes in the same build since it doesn't support monorepos)

File: packages/components/src/Listing.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      monaco.languages.setLanguageConfiguration("substance", SubstanceConfig);
+    }
+  }, [monaco]);
+  return (
+    <MonacoEditor
+      value={value}
+      width={width}
+      height={height}
+      defaultLanguage="substance"
+      // HACK
```
Comment (wodeni) [inline packages/components/src/Listing.tsx:42]: This actually typechecks. You can remove the `HACK` and cast to `any` here.

File: packages/components/src/Listing.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+};
+
+const Listing = ({
+  value,
+  width,
+  height,
+}: {
+  value: string;
+  width: string;
+  height: string;
```
Comment (wodeni) [inline packages/components/src/Listing.tsx:27]: Weird that this component is not taking in an `Env`. Don't we need that for syntax highlighting? In fact, I'm not seeing any highlighting in all examples.

File: packages/components/src/Simple.tsx [hunk 3]
```diff
@@ -3,92 +3,124 @@ import {
...
+    const compilerResult = compileTrio(this.props);
     if (compilerResult.isOk()) {
       // resample because initial sampling did not use the special sampling seed
-      const initState: PenroseState = resample(
-        await prepareState(compilerResult.value)
-      );
-      const stepped = stepUntilConvergence(initState);
+      this.penroseState = resample(await prepareState(compilerResult.value));
+    } else {
+      console.log(showError(compilerResult.error));
```
Comment (wodeni) [inline packages/components/src/Simple.tsx:38]: Not in scope: should probably do better error handling like rendering errors on the canvas directly.

File: packages/components/src/Simple.tsx [hunk 4]
```diff
@@ -3,92 +3,124 @@ import {
...
+    if (!this.props.animate) {
+      await this.converge();
+    }
+    this.renderCanvas();
+    this.timerID = window.setInterval(() => this.tick(), 1000 / 60);
   };
 
-  renderCanvas = async (state: PenroseState | undefined) => {
+  componentDidUpdate = async (prevProps: ISimpleProps) => {
+    if (
```
Comment (wodeni) [inline packages/components/src/Simple.tsx:74]: A little surprised that prop updates need to be handled this manually. Wondering if we are leveraging the React lifecycle fully.

File: packages/components/src/Demo.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+      style={{
+        display: "flex",
+        flexDirection: "row",
+        height: "100%",
+        flex: 1,
+      }}
+    >
+      <Listing value={example.sub} width={props.width} height={props.width} />
+      <div style={{ width: props.width, height: props.width }}>
+        <Simple
```
Comment (wodeni) [inline packages/components/src/Demo.tsx:38]: Nice refactoring of `Simple`! Maybe this is not "simple" anymore... we should probably rename it to `Diagram` or something.

File: packages/components/src/stories/PenrosePrograms.tsx [hunk 6]
```diff
@@ -1,12 +1,12 @@
 export const oneSet = {
```
Comment (wodeni) [inline packages/components/src/stories/PenrosePrograms.tsx:1]: Seems like there's a way to reference to static files in storybook. Perhaps we can just load files from `examples` directly: https://storybook.js.org/docs/react/configure/images-and-assets#serving-static-files-via-storybook-configuration.

File: packages/components/build.js [hunk 7]
```diff
@@ -9,6 +9,9 @@ const common = {
   define: {
```
Comment (wodeni) [inline packages/components/build.js:9]: Now that everything is vite'd. Remove this config?

File: packages/docs-site/src/pages/index.js [hunk 8]
```diff
@@ -2,7 +2,8 @@ import Link from "@docusaurus/Link";
 import useDocusaurusContext from "@docusaurus/useDocusaurusContext";
 import Layout from "@theme/Layout";
 import clsx from "clsx";
-import React from "react";
+import * as React from "react";
+import DemoWrapper from "../components/DemoWrapper";
```
Comment (wodeni) [inline packages/docs-site/src/pages/index.js:6]: Fix import: `@penrose/components`

**Review by wodeni — CHANGES_REQUESTED** [review]
Looks great to me! Everything very clean :P. The syntax highlighting issue (see below) is the only thing to look into for me.

File: packages/docs-site/src/pages/index.js (hunk 8 above)
Comment (samestep) [reply packages/docs-site/src/pages/index.js:6]: incorrect

File: packages/components/src/Listing.tsx (hunk 2 above)
Comment (samestep) [reply packages/components/src/Listing.tsx:34]: fixed

File: packages/components/src/Listing.tsx [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+  substance,
+  width,
+  height,
+}: {
+  domain: string;
+  substance: string;
+  width: string;
+  height: string;
+}) => {
+  const env = compileDomain(domain).unsafelyUnwrap();
```
Comment (wodeni) [inline packages/components/src/Listing.tsx:36]: Not in scope: same thing as the diagram component, some error handling would be good.

**Review by wodeni — APPROVED** [review]
Looks great to me! Let's merge it

**Pull request merged**